### PR TITLE
Solver Handler Transform Synchonization

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Example of solver handler implementation using `IMixedRealityControllerPoseSynchronizer`

Fixes: https://github.com/Microsoft/MixedRealityToolkit-Unity/issues/2785

(I only added a solver handler to the parabolic pointer as trying 2018.3 may have temporarily broke prefab editing for me)

### CODE REVIEW

**Comment by keveleigh** [comment]
Is this solution is to put a `ControllerPoseSynchronizer` on every solver?

**Comment by StephenHodgson** [comment]
>Is this solution is to put a ControllerPoseSynchronizer on every solver?

No, this doesn't make that a requirement

**Comment by keveleigh** [comment]
But to track controllers, it is?

**Comment by StephenHodgson** [comment]
> But to track controllers, it is?

Not the Concrete type, but the IMixedRealityControllerPoseSynchronizer, yes. or else how would you get any data about any controller?

**Comment by keveleigh** [comment]
Using a controller pose synchronizer for every solver that needs to track a controller does not fulfill the acceptance criteria for #2785 of 

>Doesn't add another global listener to the scene

**Comment by StephenHodgson** [comment]
> Doesn't add another global listener to the scene

It doesn't. It's using the ones we're already using.

**Comment by keveleigh** [comment]
> It's using the ones we're already using.

If I create a new prefab that adds a cube to the front of my right controller via solver, I will not need to add a new controller pose synchronizer to it?

**Comment by StephenHodgson** [comment]
Nope, not as long as you get your right controller's `IMixedRealityControllerPoseSynchronizer` reference

**Comment by keveleigh** [comment]
Where can I get that reference?

**Comment by StephenHodgson** [comment]
[It gets it for you](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/2789/files#diff-838f7d94c2a0c39be3bd3932d0b225f3R146), if the handler is already attached to a game object that has a `IMixedRealityControllerPoseSynchronizer`

**Comment by StephenHodgson** [comment]
Please feel free to take this and change it up to better fit what your needs are, open a PR on my branch, or open a new PR with your own take that tries to synchronize the pose in this way.

**Comment by keveleigh** [comment]
To clarify, back to my scenario a few comments up, if I have a brand new cube prefab and a add a solver to it, where do I get an existing reference to a pose synchronizer?

File: Assets/MixedRealityToolkit-SDK/Features/Utilities/Solvers/InBetween.cs [hunk 1]
```diff
@@ -25,26 +25,10 @@ public float PartwayOffset
...
-            set
-            {
-                trackedObjectForSecondTransform = value;
-                if (secondSolverHandler != null)
-                {
-                    secondSolverHandler.TrackedObjectToReference = value;
-                }
-            }
-        }
+        public TrackedObjectType TrackedObjectForSecondTransform => secondSolverHandler.TrackedObjectType;
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/Utilities/Solvers/InBetween.cs]: You can still set any transform to track, this is just a read only value of the currently tracked transform's type.

**Comment by StephenHodgson** [comment]
Added a DNM as this is just an example.

File: Assets/MixedRealityToolkit-SDK/Features/Utilities/Solvers/InBetween.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ecnassianer) [reply Assets/MixedRealityToolkit-SDK/Features/Utilities/Solvers/InBetween.cs]: Ah, right. "secondTransformOverride" didn't show up in the diff, so I got the two mixed up. 

But I guess that prompts a similar question. If this becomes read only, how does gameplay logic tell the solver to start tracking a motion controller again? It has to find the proper motion controller in the scene and override the TransformTarget? That seems like an undesirable complication.

[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/Utilities/Solvers/InBetween.cs]: I can see how that would be useful.

**Pull request closed without merging**